dataland-frontend/src/components/general/SelectReportingPeriodDialog.vue

```diff
@@ -43,12 +53,31 @@ export default defineComponent({
       if (this.mapOfReportingPeriodToActiveDataset) {
         const sortedReportingPeriodMetaInfoPairs = Array.from(
           (this.mapOfReportingPeriodToActiveDataset as Map<string, DataMetaInformation>).entries(),
-        ).sort((firstEl, secondEl) => compareReportingPeriods(firstEl[0], secondEl[0]));
+        ).sort((firstElement, secondElement) => compareReportingPeriods(firstElement[0], secondElement[0]));
         for (const [key, value] of sortedReportingPeriodMetaInfoPairs) {
+          const answeredDataRequestId = this.answeredDataRequests
+            ?.filter((answeredDataRequest: StoredDataRequest) => {
+              return answeredDataRequest.reportingPeriod == key;
+            })
+            .map((answeredDataRequest: StoredDataRequest) => {
+              return answeredDataRequest.dataRequestId;
+            });
+          let isClickable;
+          if (this.actionOnClick == ReportingPeriodTableActions.EditDataset) {
+            isClickable = true;
+          } else if (answeredDataRequestId) {
+            isClickable = answeredDataRequestId.length > 0;
+          } else {
+            isClickable = false;
+          }
+
```

Comment:
the last two if statements could be simplified to the following:

isClickable = answeredDatarequest && answeredDataRequestId.length > 0.

in case the prop this.answeredDataRequest is always defined, the first part of the && statement is obsolete, because even if the filter doesn't find a reporting period in line 60, answeredDataRequestId will be an empty array.
then             isClickable = answeredDataRequestId.length > 0 would be enough without checking if answeredDataRequestId is non-null.

Reply:
Nice solution. I changed it. Unfortunately the prop this.answeredDataRequest is not always defined, so we need the is-non-null statement